### PR TITLE
on graph serialization, persisting smaller chunks of the graph straight ...

### PR DESCRIPTION
**Pull request by eishay**
...to disk without needing to allocate large string representation of the set of edges.

note that the serialization format had changed and the reader is backward compatible.

### CODE REVIEW

**Comment by eishay** [comment]
@ymatsuda @leogrim

File: kifi-backend/modules/graph/app/com/keepit/graph/simple/MutableVertex.scala [hunk 1]
```diff
@@ -33,11 +35,43 @@ class MutableVertex(var data: VertexDataReader, val outgoingEdges: MutableOutgoi
...
     if (incomingEdges.edges(component).isEmpty) { incomingEdges.edges -= component }
   }
+
+  /**
+   * UTF, Int, (Long, UTF)
+   */
+  def write(out: ObjectOutputStream): Unit = {
+    out.writeUTF(VertexDataReader.writes.writes(data).toString)
+    out.writeInt(outgoingEdges.edges.foldLeft(0)((count, edges) => count + edges._2.size))
+    outgoingEdges.edges foreach {
```
Comment (ymatsuda) [inline kifi-backend/modules/graph/app/com/keepit/graph/simple/MutableVertex.scala:45]: `outgoingEdges.edges` is mutable. There is a slight chance that the count becomes inconsistent.

Comment (eishay) [inline kifi-backend/modules/graph/app/com/keepit/graph/simple/MutableVertex.scala:45]: @leogrim are we not locking writes when persisting the graph?
If not, I can clone it to an immutable list.

Comment (leogrim) [inline kifi-backend/modules/graph/app/com/keepit/graph/simple/MutableVertex.scala:45]: Yes we do, see `GraphManagerActor`, calls `graph.update` and `graph.backup` are synchronous there.

**Comment by leogrim** [comment]
:+1: Thanks!

**Pull request merged**